check-version.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+$majorVersionInUse = PHP_MAJOR_VERSION;
+$fullVersionInUse = phpversion();
+echo sprintf(
+    "Using PHP %s\n",
+    $fullVersionInUse
+);
+
+$url = 'https://www.php.net/releases/?json&version=' . $majorVersionInUse;
+$releaseJson = file_get_contents($url);
+$releaseData = json_decode($releaseJson, true);
+$latestRelease = $releaseData['version'];
+echo sprintf(
+    "Latest PHP %s release: %s\n",
+    $majorVersionInUse,
+    $latestRelease
+);
+
+$isUsingLatestRelease = ($fullVersionInUse === $latestRelease);
+echo sprintf(
+    "This %s using the latest PHP %s release\n",
+    $isUsingLatestRelease ? 'IS' : 'is NOT',
+    $majorVersionInUse
+);
+
+exit($isUsingLatestRelease ? 0 : 1);
```

Comment:
Right, because bash uses 0 for success and non-zero for failure. The exit() will give that code back to what called the php script. My issue is whether this accomplishes the intent. Won't the build fail when it isn't the latest release? So, is the intent that the build fail will send an email to the appropriate people and they will update the PHP used, and this rerunning will build pass?

Reply:
Close. It's not a build that will fail; it's when we run this Docker image that it will "fail" (exit 1) or "succeed" (exit 0).

My hope is to find a way to run this on a schedule in AWS, and if it exits with an error (exit 1) have it email us. Once we update the base image, a PR will be submitted on this repo to update it. Once we update this repo, the next daily run will succeed, resulting in us getting an email every day that we're using an old version of PHP.

Reply:
docker run --rm silintl/php7-apache-updater:develop
This generates the exit 1 case.

Locally editing the FROM to
FROM php:7.4.19
and then building and
docker run {image number}
gives an exit 0.